impl/src/main/java/org/jboss/weld/Container.java

```diff
@@ -40,6 +42,8 @@
 
     private static Singleton<Container> instance;
 
+    private static AtomicReference<Environment> environmentReference = new AtomicReference<>(null);
```

Comment:
Actually, this should be a constant, i.e. `private static final AtomicReference<Environment> ENV_REFERENCE = new AtomicReference<>();`